src/dsl.jl

```diff
@@ -713,34 +717,41 @@ function read_equations_options(options, variables_declared; requiredec = false)
     # Loops through all equations, checks for lhs of the form `D(X) ~ ...`.
     # When this is the case, the variable X and differential D are extracted (for automatic declaration).
     # Also performs simple error checks.
-    vars_extracted = Vector{Symbol}()
+    vars_extracted = OrderedSet{Union{Symbol, Expr}}()
```

Comment:
species and parameters in the corresponding cases uses OrderedSet{Union{Symbol, Expr}}(), so figured we should use it here too.